code/datums/components/jousting.dm

```diff
@@ -0,0 +1,75 @@
+/datum/component/jousting
+	var/current_direction = NONE
+	var/max_tile_charge = 5
+	var/current_tile_charge = 0			//dir = tiles
+	var/movement_reset_tolerance = 2			//deciseconds
+	var/unmounted_damage_boost_per_tile = 0
+	var/unmounted_knockdown_chance_per_tile = 0
+	var/unmounted_knockdown_time = 0
+	var/mounted_damage_boost_per_tile = 2
+	var/mounted_knockdown_chance_per_tile = 20
+	var/mounted_knockdown_time = 20
+	var/requires_mob_riding = TRUE			//whether this only works if the attacker is riding a mob, rather than anything they can buckle to.
+	var/requires_mount = TRUE				//kinda defeats the point of jousting if you're not mounted but whatever.
+	var/mob/current_holder
+	var/datum/component/redirect/listener
+	var/current_timerid
+
+/datum/component/jousting/Initialize()
+	if(!isitem(parent))
+		. = COMPONENT_INCOMPATIBLE
+		stack_trace("Warning: Jousting component incorrectly applied to invalid parent type [parent.type]")
+	RegisterSignal(COMSIG_ITEM_EQUIPPED, .proc/on_equip)
+	RegisterSignal(COMSIG_ITEM_DROPPED, .proc/on_drop)
+	RegisterSignal(COMSIG_ITEM_ATTACK, .proc/on_attack)
+
+/datum/component/jousting/Destroy()
+	QDEL_NULL(listener)
+	return ..()
+
+/datum/component/jousting/proc/on_equip(mob/user, slot)
+	QDEL_NULL(listener)
+	current_holder = user
+	listener = new(user, COMSIG_MOVABLE_MOVED, CALLBACK(src, .proc/mob_move))
+
+/datum/component/jousting/proc/on_drop(mob/user)
+	QDEL_NULL(listener)
+	current_holder = null
+	current_direction = NONE
+	current_tile_charge = 0
+
+/datum/component/jousting/proc/on_attack(mob/living/target, mob/user)
+	var/current = current_tile_charge
+	var/target_buckled = target.buckled ? TRUE : FALSE			//we don't need the reference of what they're buckled to, just whether they are.
+	if((requires_mount && ((requires_mob_riding && !ismob(user.buckled)) || (!user.buckled))) || !current_direction)
+		return
+	var/turf/target_turf = get_step(user, current_direction)
+	if(target in RANGE_TURFS(1, target_turf))
+		var/knockdown_chance = (target_buckled? mounted_knockdown_chance_per_tile : unmounted_knockdown_chance_per_tile) * current
+		var/knockdown_time = (target_buckled? mounted_knockdown_time : unmounted_knockdown_time)
+		var/damage = (target_buckled? mounted_damage_boost_per_tile : unmounted_damage_boost_per_tile) * current

```

Comment:
Forgive me if I'm missing something here, but is this extra damage even used anywhere? It seems to be defined and the message given but it never actually applied.